docs/developer-docs/backend/rust/intercanister.mdx

```diff
@@ -8,169 +8,72 @@ import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
 
-Now that you've taken a look at your canisters, let's deploy them.
+We will next show how to use *bounded wait* calls instead in such cases. These calls don't guarantee that the response will be delivered, which is why we also refer to the as *best-effort response calls*, but in return don't block the caller from upgrading, and they also tolerate high system load better than unbounded wait calls. See the section on [inter-canister calls](/docs/current/developer-docs/smart-contracts/advanced-features/async-code) for more information on best-effort vs. guaranteed response calls.
 
-Open a terminal window on your local computer, if you don’t already have one open.
+## ICRC-1 transfers: bounded wait calls
 
-Then run the commands:
+We will now allow our wallet to transfer tokens on an arbitrary ICRC-1 ledger instead of just the ICP ledger. Since we in general can't trust an arbitrary ICRC-1 ledger, and we want to ensure that our canister can always be upgraded, we will use bounded wait calls.
 
-```bash
-dfx start --clean --background
-dfx deploy
-```
+Ledgers generally charge fees for transfers. While this fee is fixed for the ICP ledger, it may vary for other ledgers. Thus, we start with an example of how to determine the required fee.
 
-## Making inter-canister calls
+### Learning the transfer fee
 
-First, let's subscribe to a topic. For example, to subscribe to the "Apples" topic, use the command:
+Querying the transfer fee does not change the ledger state. Thus, it's simple to retry in case that it fails, and the code below implements basic retries.
 
-```bash
-dfx canister call subscriber setup_subscribe '(principal "<INSERT_PUBLISHER_PRINCIPAL_HERE>", "Apples")'
+```rust reference
+https://github.com/oggy-dfin/icc_rust_docs/blob/34f59ddae9fcc70173fc21927a4279757f93c51a/src/icc_rust_docs_backend/src/lib.rs#L83-L134
 ```
 
-Then, to publish a record to the "Apples" topic, use the command:
+As noted in the example, the code after the first call executes in a different callback. See the sections on [inter-canister calls and async code](/docs/current/developer-docs/smart-contracts/advanced-features/async-code), [properties of call execution](/docs/current/developer-docs/security/security-best-practices/inter-canister-calls) and [security best practices](docs/current/developer-docs/security/security-best-practices/inter-canister-calls) to understand potential security implications for your application when using inter-canister calls.
 
-```bash
-dfx canister call publisher publish '(record { "topic" = "Apples"; "value" = 2 })'
-```
+### Transferring tokens
 
-Then, you can query and receive the subscription record value with the command:
+When transferring tokens (or performing other updates) using bounded wait messages, we need to handle the unknown state case. For ICRC-1 transfers, we can make use of the built-in deduplication feature of ICRC-1 ledgers and retry the call.
```

Comment:
Should we maybe link to an explanation of the semantics of this deduplication? Or should we maybe briefly explain it here?

Reply:
This is now moved to the ICRC-1 page and the deduplication semantics are linked to there